base/float.jl

```diff
@@ -688,22 +688,25 @@ function hash(x::Real, h::UInt)
     den_z = trailing_zeros(den)
     den >>= den_z
     pow += num_z - den_z
-
-    # handle values representable as Int64, UInt64, Float64
+    # If the real is an Int64, UInt64, or Float64, hash as those types.
+    # To be an Integer the denominator must be 1 and the power must be positive.
+    # To be a Float64
```

Comment:
Something missing here?

Reply:
oops. That got moved to the comment on line 704